alacritty/src/display/mod.rs

```diff
@@ -778,6 +779,47 @@ impl Display {
         });
     }
 
+    /// Draw the hyperlink target URI preview of which are currently active.
+    fn draw_active_hyperlink_preview(&mut self, config: &Config, size_info: &SizeInfo) {
+        // Currently we simply stack vi-highlighted and cursor-pointed URL if both exist.
+        for (idx, uri) in self
+            .highlighted_hint
+            .iter()
+            .chain(&self.vi_highlighted_hint)
+            .filter_map(|m| match m {
+                HintMatch::Hyperlink { hyperlink, .. } => Some(hyperlink.uri()),
+                _ => None,
+            })
+            .enumerate()
+        {
+            let glyph_cache = &mut self.glyph_cache;
+
+            let point = Point::new(size_info.screen_lines().saturating_sub(1 + idx), Column(0));
+            let fg = config.ui_config.colors.information_bar_foreground();
+            let bg = config.ui_config.colors.information_bar_background();
+
+            let mut text = String::with_capacity(uri.len());
+            let mut text_len = 0usize;
+            // Leave room for the ellipsis character.
```

Comment:
Curious if we should maybe just define this ellipsis character as a constant and then use `unicode_width` or `len` or something to build the logic on dynamically. In theory if someone would want to change the character for some reason that would make it much easier then (to `...` for example).

We do the same thing with search I believe.